lib/fastlane/actions/resize_app_icon.rb

```diff
@@ -0,0 +1,120 @@
+module Fastlane
+  module Actions
+    # ResizeAction is based on work created and Copyright (c) 2014 Erik Sundin
+    # Original source may be found under the MIT license at https://github.com/eriksundin/appicon
+    class ResizeAppIconAction < Action
+      def self.run(params)
+        # fastlane will take care of reading in the parameter and fetching the environment variable:
+        Helper.log.info "Path to source icon: #{params[:iconPath]}"
+        Helper.log.info "Path to Images.xcassets: #{params[:assetsPath]}"
+
+        if params[:iconPath] and params[:assetsPath] and !params[:iconPath].empty? and !params[:assetsPath].empty?
```

Comment:
Please do any kind of validation in the `available_options` method and not here. Also, if a value is required, you'll get that for free with `fastlane`